src/dill/gauge-proxy-v2.sol

```diff
@@ -0,0 +1,2161 @@
+
+        uint256 j = 0;
+        for (uint256 i = 0; i < tokensLength; i++) {
+            if (_tokens[i] != _token) {
+                newTokenArray[j] = _tokens[i];
+                j++;
+            }
+        }
+
+        _tokens = newTokenArray;
+    }
+
+    // Sets MasterChef PID
+    function setPID(uint256 _pid) external {
+        require(msg.sender == governance, "!gov");
+        require(pid == 0, "pid has already been set");
+        require(_pid > 0, "invalid pid");
+        pid = _pid;
+    }
+
+    // Deposits mDILL into MasterChef
+    function deposit() public {
+        require(pid > 0, "pid not initialized");
+        IERC20 _token = TOKEN;
+        uint256 _balance = _token.balanceOf(address(this));
+        _token.safeApprove(address(MASTER), 0);
+        _token.safeApprove(address(MASTER), _balance);
+        MASTER.deposit(pid, _balance);
+    }
+
+    // Fetches Pickle
+    function collect() public {
+        (uint256 _locked, ) = MASTER.userInfo(pid, address(this));
+        MASTER.withdraw(pid, _locked);
+        deposit();
+    }
+
+    function length() external view returns (uint256) {
+        return _tokens.length;
+    }
+
+    function distribute(uint256 _start, uint256 _end) external {
+        require(_start < _end, "GaugeProxyV2: bad _start");
+        require(_end <= _tokens.length, "GaugeProxyV2: bad _end");
+        require(
+            msg.sender == governance,
+            "GaugeProxyV2: only governance can distribute"
+        );
+
+        uint256 currentId = getCurrentPeriodId();
+        require(
+            distributionId < currentId,
+            "GaugeProxyV2: all period distributions complete"
+        );
+
+        uint256 periodToUse = 0;
+        if (periodForDistribute[distributionId] == 0) {
+            // If period does not exist means no votes in this period
+            // Use previous period's votes and update dist. period
+            periodToUse = periodForDistribute[distributionId - 1];
+            periodForDistribute[distributionId] = periodForDistribute[
+                distributionId - 1
+            ];
+        } else {
+            periodToUse = periodForDistribute[distributionId];
+        }
+
+        collect();
+        int256 _balance = int256(PICKLE.balanceOf(address(this)));
+        int256 _totalWeight = totalWeight[periodToUse];
+
+        if (_balance > 0 && _totalWeight > 0) {
+            for (uint256 i = _start; i < _end; i++) {
+                if (distributed[distributionId][i]) {
+                    continue;
+                }
+                address _token = _tokens[i];
+                address _gauge = gauges[_token];
+                int256 _reward = (_balance * weights[periodToUse][_token]) /
+                    _totalWeight;
+                if (_reward > 0) {
+                    uint256 reward_ = uint256(_reward);
+                    uint256[] memory rewardArr = new uint256[](1);
```

Comment:
I see that the gauges are meant to accomodate multiple types of rewards, but this `distribute` function never allows for the other rewards to be realized (as there's an `onlyDistribution` modifier on the gauge contracts)

Reply:
Check for onlyDistribution modifier

Reply:
Add limit of 8 reward tokens